File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.java
class Solution {
    int maxSubarraySum(int[] arr) {
        // Code here
        int currSum = arr[0];
        int sum = arr[0];
        
        for(int i=1; i<arr.length; i++){
            currSum = Math.max(arr[i], currSum + arr[i]);
            sum = Math.max(currSum, sum);
        }
        return sum;
        
    }
}

File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.py
class Solution:
    def maxSubarraySum(self, arr):
        # Code here
        maxSum =arr[0]
        currMax =arr[0]
        
        for i in range(1, len(arr)):
            currMax = max(arr[i], currMax + arr[i])
            
            maxSum = max(maxSum, currMax) 
        
        return maxSum
        
        
        